dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/TracingInterceptor.kt

```diff
@@ -40,29 +42,60 @@ import okhttp3.Response
  */
 class TracingInterceptor : Interceptor {
 
+    private val localTracerReference: AtomicReference<Tracer> = AtomicReference()
+
     // region Interceptor
 
     /** @inheritdoc */
     override fun intercept(chain: Interceptor.Chain): Response {
-
-        if (GlobalTracer.isRegistered()) {
-            val tracer = GlobalTracer.get()
+        val tracer = resolveTracer()
+        return if (tracer != null) {
             val span = tracer.buildSpan("okhttp.request")
                 .start()
+            updateAndProceedSafely(chain, tracer, span)
+        } else {
+            chain.proceed(chain.request())
+        }
+    }
+
+    // endregion
+
+    // region Internal
 
-            return updateAndProceedSafely(chain, tracer, span)
+    private fun resolveTracer(): Tracer? {
+        if (GlobalTracer.isRegistered()) {
+            // clear the localTracer reference if any
+            localTracerReference.set(null)
+            return GlobalTracer.get()
         } else {
+            if (!TracesFeature.initialized.get()) {
+                devLogger.w(
+                    "You added the TracingInterceptor to your OkHttpClient " +
+                            "but you did not enable the TracesFeature."
+                )
+                return null
+            }
+
+            // we check if we already have a local tracer if not we instantiate one
+            return resolveLocalTracer()
```

Comment:
So, the local tracer will add exactly the same data to intercepted requests as `GlobalTracer.get()` would do if registered?

Reply:
yes...is just that we are creating for him if he forgets